### DISCUSSION TITLE
Can we set defaultValue depend on locale?

### DISCUSSION THREAD

**Discussion by Duckinm** [payloadcms/payload#273] [Feature Requests & Ideas]
I want fields to automatically update the default value by checking locale

Is it a possible way to do that?

**Comment by jmikrut** [comment]
Dang, this is a super good question.

It would likely need to be a feature request. We could allow the `defaultValue` field to accept either a value, or a function that returns a value—wherein you could return a value based on other fields' data, locale, etc.

Alternatively, right now, you could use hooks (say a field `beforeChange` hook) that set a default value if no value is passed. This would be a stop-gap solution that would not have any effects on the admin UI, but after a document was saved, the default value would then show up.

In the interim, I've switched this discussion category to Feature Request so we can keep this idea around!

**Comment by bratvanov** [comment]
This feature request can be closed, since `defaultValue` as a function has been supported for a while:
https://payloadcms.com/docs/fields/overview#default-values

> Here is an example of a `defaultValue` function:
> ```ts
> import type { Field } from 'payload'
> 
> const translation: {
>   en: 'Written by'
>   es: 'Escrito por'
> }
> 
> export const myField: Field = {
>   name: 'attribution',
>   type: 'text',
>   defaultValue: ({ user, locale, req }) =>
>     `${translation[locale]} ${user.name}`,
> }
> ```
>
> Tip: You can use async `defaultValue` functions to fill fields with data from API requests or Local API using `req.payload`.
